worksheet_A/TerminalHacking.cpp

```diff
@@ -27,20 +39,48 @@ int main()
 
 	// Fill the set with more words
 	// Using a set for options guarantees that the elements are all different
-	while (options.size() < numberOfWords)
-	{
+	while (options.size() < numberOfWords){
 		std::string word = words.getRandomWord();
 		options.insert(word);
 	}
 
 	// Display all words
-	for each (std::string word in options)
-	{
+	for each (std::string word in options){
 		std::cout << word << std::endl;
 	}
 
-	// TODO: implement the rest of the game
+	
+	std::string guess = "";
+	int Likeness = 0;
+	int guessremain = 0;
 
-    return 0;
-}
+	while (guessremain < 4){
+		std::cout << "\nPlease enter your guess\n";
+		std::transform(cout.begin(), cout.end(),cout.begin(), ::toupper);
+		std::cin >> guess;
+		std::set<std::string>::iterator Counter = options.find(guess);
+		
+		if (Counter == options.end()){
+			std::cout << "Invalid Guess." << std::endl;
+			continue;
+		}
+
+		else{
+			Likeness = GetLikeness(guess, secret);
+			std::cout << "\nyou have a likeness of " << Likeness << " out of " << WordLength << "\n";
 
+			if (Likeness == WordLength){
+				std::cout << "\nyou won\n";
+				exit(0);
+			}
+			
+			else{
+				guessremain++;
```

Comment:
Confusing naming -- I would assume that `guessremain` is the number of guesses remaining, however it is actually the number of guesses used.